cmd/webhook/main.go

```diff
@@ -0,0 +1,304 @@
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func newApp() *cli.App {
+	flags := &Flags{
+		loggingConfig:     flags.NewLoggingConfig(),
+		featureGateConfig: flags.NewFeatureGateConfig(),
+	}
+	cliFlags := []cli.Flag{
+		&cli.StringFlag{
+			Name:        "tls-cert-file",
+			Usage:       "File containing the default x509 Certificate for HTTPS. (CA cert, if any, concatenated after server cert).",
+			Destination: &flags.certFile,
+			Required:    true,
+		},
+		&cli.StringFlag{
+			Name:        "tls-private-key-file",
+			Usage:       "File containing the default x509 private key matching --tls-cert-file.",
+			Destination: &flags.keyFile,
+			Required:    true,
+		},
+		&cli.IntFlag{
+			Name:        "port",
+			Usage:       "Secure port that the webhook listens on",
+			Value:       443,
+			Destination: &flags.port,
+		},
+	}
+	cliFlags = append(cliFlags, flags.loggingConfig.Flags()...)
+	cliFlags = append(cliFlags, flags.featureGateConfig.Flags()...)
+
+	app := &cli.App{
+		Name:            "webhook",
+		Usage:           "webhook implements a validating admission webhook complementing a DRA driver plugin.",
+		ArgsUsage:       " ",
+		HideHelpCommand: true,
+		Flags:           cliFlags,
+		Before: func(c *cli.Context) error {
+			if c.Args().Len() > 0 {
+				return fmt.Errorf("arguments not supported: %v", c.Args().Slice())
+			}
+			return flags.loggingConfig.Apply()
+		},
+		Action: func(c *cli.Context) error {
+			server := &http.Server{
+				Handler: newMux(),
+				Addr:    fmt.Sprintf(":%d", flags.port),
+			}
+			klog.Info("starting webhook server on", server.Addr)
+			return server.ListenAndServeTLS(flags.certFile, flags.keyFile)
+		},
+	}
+
+	return app
+}
+
+func newMux() *http.ServeMux {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/validate-resource-claim-parameters", serveResourceClaim)
+	mux.HandleFunc("/readyz", func(w http.ResponseWriter, req *http.Request) {
+		_, err := w.Write([]byte("ok"))
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	})
+	return mux
+}
+
+func serveResourceClaim(w http.ResponseWriter, r *http.Request) {
+	serve(w, r, admitResourceClaimParameters)
+}
+
+// serve handles the http portion of a request prior to handing to an admit
+// function.
+func serve(w http.ResponseWriter, r *http.Request, admit func(admissionv1.AdmissionReview) *admissionv1.AdmissionResponse) {
+	var body []byte
+	if r.Body != nil {
+		data, err := io.ReadAll(r.Body)
+		if err != nil {
+			klog.Error(err)
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		body = data
+	}
+
+	// verify the content type is accurate
+	contentType := r.Header.Get("Content-Type")
+	if contentType != "application/json" {
+		msg := fmt.Sprintf("contentType=%s, expected application/json", contentType)
+		klog.Error(msg)
+		http.Error(w, msg, http.StatusUnsupportedMediaType)
+		return
+	}
+
+	klog.V(2).Infof("handling request: %s", body)
```

Comment:
It makes sense to not log the entire request body, but at most N bytes (I often pick something like 100 or 500). 

For not accidentally logging a megabyte of data in case something goes pearshaped. :pear:

Reply:
Include in the follow-up as well?